knack/config.py

```diff
@@ -47,36 +47,126 @@ def __init__(self, config_dir=None, config_env_var_prefix=None, config_file_name
         self._env_var_format = '{}{}'.format(env_var_prefix, '{section}_{option}')
         self.defaults_section_name = CLIConfig._CONFIG_DEFAULTS_SECTION
         self.config_parser.read(self.config_path)
+        self.config_file_chain = []
+        current_dir = os.getcwd()
+        config_dir_name = os.path.basename(self.config_dir)
+        while current_dir:
+            current_config_dir = os.path.join(current_dir, config_dir_name)
+            if (os.path.normcase(os.path.normpath(current_config_dir)) ==
+                    os.path.normcase(os.path.normpath(self.config_dir))):
+                break
+            if os.path.isdir(current_config_dir):
+                self.config_file_chain.append(_ConfigFile(current_config_dir,
+                                                          os.path.join(current_config_dir, configuration_file_name)))
+            if current_dir == os.path.dirname(current_dir):
```

Comment:
I assume this means we are getting to the root drive? if yes, I suggest call out in the comment

Reply:
Added comments.